src/UI/Components/Diagram/routers.ts

```diff
@@ -0,0 +1,65 @@
+import { dia, g, routers } from "@clientio/rappid";
+import { RouterOptions } from "./interfaces";
+
+const DEFAULT_PADDING = 30;
+
+export const routerNamespace = { ...routers };
+
+function getOutsidePoint(
+  bbox: g.Rect,
+  angle: number,
+  anchor: g.Point,
+  padding: number
+) {
+  const ref = anchor.clone();
+  const center = bbox.center();
+  if (angle) ref.rotate(center, angle);
+  const point = new g.Point(bbox.x, ref.y);
+  if (point.equals(anchor)) {
+    point.x--;
+    padding--;
+  }
+  point.move(ref, ref.x < center.x ? padding : -bbox.width - padding);
```

Comment:
To be honest I would wait with documenting router logic as it will be changed at some point as it doesn't currently fit the design, that file is part from demo and will serve as a base for further modification and I don't know which parts of the code from that will stay. 
That part move the "anchor" point of connections between the shapes into the middle of selected parameter(https://www.jointjs.com/demos/database it shown on demo) and that's something that we won't have in the final version, but I didn't change it as it exceeds goal of the milestone. I hope that understandable.